### PR TITLE
Kafka

### PR DESCRIPTION
**Pull request by 3epi**
adding kafka to the docker compose 
producer.py code 
dockerfile for the producer.py 

** producer still doesn't work there is an issue in the containers communication

### CODE REVIEW

File: docker-compose.yml
Comment (sorushsaghari) [inline docker-compose.yml:1]: use bitnami kafka.

File: producer.py
Comment (sorushsaghari) [inline producer.py:1]: read your configuration from env, file or args. and set the config in you docker compose file.
add a for loop in code to generate messages(end less).

**Review by sorushsaghari — CHANGES_REQUESTED** [review]

File: docker-compose.yml
Comment (3epi) [reply docker-compose.yml:1]: done this and added environment variables , still connection between containers doesn't work and neither of produer , consumer  can find the kafka broker

File: producer.py
Comment (3epi) [reply producer.py:1]: everything works fine now, just have  to change consumer to save the data in a database. and the producer to make random data not just one text over and over again

**Comment by 3epi** [comment]
everything works fine and the randomly generated strings are saved in a mongoDB database.

**Comment by 3epi** [comment]
just saw the readme in github :D tomorrow will fix that so it will look better.

File: consumer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM docker.arvancloud.ir/python:3.11
+
+RUN pip3 install kafka-python six pymongo
```
Comment (sorushsaghari) [inline consumer/Dockerfile]: find a solution to add dependecy manegement like version tracking. do not call pip install packages like this.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 services:
   prometheus:
-    image: prom/prometheus:latest
+    image: docker.arvancloud.ir/prom/prometheus:latest
```
Comment (sorushsaghari) [inline docker-compose.yml]: it is a besta practice to specify version of your dependency or image. use this practice and add it to your readme why you have to do this.

Comment (3epi) [reply docker-compose.yml]: fixed

File: consumer/Dockerfile (hunk 1 above)
Comment (3epi) [reply consumer/Dockerfile]: fixed

**Review by sorushsaghari — APPROVED** [review]
good job. lets have these TODOs for next time:
- config management
- batch insertion
- automated deploy with ansible
- a web api as producer :)

**Pull request merged**